runtime/test/recipe-test.js

```diff
@@ -39,4 +39,59 @@ describe('recipe', function() {
     recipe.handles.forEach(handle => assert.isTrue(options.errors.has(handle)));
     options.errors.has(recipe.slots[1]);
   });
+  it('validate handle connection types', async () => {
+    let manifest = await Manifest.parse(`
+        schema MyType
+        schema MySubType extends MyType
+        schema OtherType
+        particle P1
+          P1(in MyType inMy)
+        particle P2
+          P2(out MyType inMy)
```

Comment:
Rename to `outMy` just for clarity? Or just rename all of the particle params to something generic like `dummy` since it doesn't look like they really matter.

Reply:
done